components/TopBar.tsx

```diff
@@ -0,0 +1,40 @@
+import { AppBar, Box, Button, Toolbar, Typography, useTheme } from '@mui/material';
+import React, { useContext } from 'react';
+import { ColorModeToggleSwitch } from './ColorModeToggleSwitch';
+import { ColorModeContext } from '../utils/ColorModeContext';
+
+// declare react component for top bar
+const TopBar: React.FC = () => {
+  const colorMode = useContext(ColorModeContext);
+  const theme = useTheme();
+  return (
+    <AppBar position="static">
+      <Toolbar>
+        <Box sx={{ flexGrow: 1 }}>
```

Comment:
Top bar now uses two boxes, left box is allowed to grow, right box is not.
The right box currently just contains the theme toggle.